grails-app/controllers/au/org/ala/biocache/hubs/OccurrenceController.groovy

```diff
@@ -460,7 +460,9 @@ class OccurrenceController {
                         skin: grailsApplication.config.skin.layout
                 ])
             } else {
-                flash.message = "No record found with id: ${id}"
+                if (record != null) {
```

Comment:
What about `record == {}` (not the same as `null`?) - it won't be caught by either the `if` or in this `else` block?

Reply:
Hi Nick,

            if (code == 200) {
                return JSON.parse(conn.getInputStream(), "UTF-8")
            } else if (code == 404) {
                return new JSONObject()
            } else {
                return null
            }

record can only be 
1.  valid (non-empty) json object which goes to normal page
2. empty json or null, both comes to `else`, when not null, means it's 404 so `flash.message` is set.